test/v1/service.go

```diff
@@ -201,16 +201,38 @@ func PatchServiceRouteSpec(t testing.TB, clients *test.Clients, names test.Resou
 	})
 }
 
+// WaitForServiceLatestRevisionForConformanceTest takes a revision in through names and compares it to the current state of LatestCreatedRevisionName in Service.
+// Once an update is detected in the LatestCreatedRevisionName, the function waits for the created revision to
+//   1. be set in LatestReadyRevisionName
+// before returning the name of the revision.
+func WaitForServiceLatestRevisionForConformanceTest(clients *test.Clients, names test.ResourceNames) (string, error) {
+	return waitForServiceLatestRevision(clients, names, false)
+}
+
 // WaitForServiceLatestRevision takes a revision in through names and compares it to the current state of LatestCreatedRevisionName in Service.
-// Once an update is detected in the LatestCreatedRevisionName, the function waits for the created revision to be set in LatestReadyRevisionName
+// Once an update is detected in the LatestCreatedRevisionName, the function waits for the created revision to
+//   1. become routing active
+//   2. be set in LatestReadyRevisionName
 // before returning the name of the revision.
+// NOTE: Do NOT use this for conformance tests.
 func WaitForServiceLatestRevision(clients *test.Clients, names test.ResourceNames) (string, error) {
+	return waitForServiceLatestRevision(clients, names, true)
+}
+
+func waitForServiceLatestRevision(clients *test.Clients, names test.ResourceNames, checkLabel bool) (string, error) {
```

Comment:
What do you think about alternatively removing the label check from WaitForServiceLatestRevision entirely and instead adding a separate WaitForRevisionNameLabel method which we only call in upgrade tests / where it's needed? That way it'd be a lot clearer (I think) what was actually happening, and we could avoid a method with a "ConformanceTest" suffix.

Reply:
ah, sorry, I misread what was going on here! It's actually the revision routing label which we're wanting to skip, so this doesn't work 🤔. Still I'd really like it if we could avoid a method suffixed "ConformanceTest". Is there a way to achieve the goals of being resilient during upgrade that's both conformant _and_ works during upgrades? If so could we use that, if not could we add one to conformance? Failing that, at least maybe we could have something like `WaitForServiceLatestRevisionWithoutRouteLabelCheck` to be clear how the methods actually differ (but obviously all these names are super long which isn't ideal).

Reply:
I was hesitating between `WaitForServiceLatestRevisionWithoutLabelCheck` and `WaitForServiceLatestRevisionForConformanceTest`. I like `WaitForServiceSingleLatestRevisionReady`. Changed.

Reply:
oh actually it's `WaitForServiceSingleLatestRevision` to be consistent.